test-suites/utils/excuteStory.ts

```diff
@@ -0,0 +1,146 @@
+import { Wallet } from "@ethersproject/wallet";
+import {
+  expectedReserveDataAfterInvestMoneyPool,
+  expectedUserDataAfterInvestMoneyPool,
+  expectedReserveDataAfterWithdrawMoneyPool,
+  expectedUserDataAfterWithdrawMoneyPool,
+} from "../../test/utils/Expect";
+import { getTimestamp } from "../../test/utils/Ethereum";
+import { ethers } from 'hardhat';
+import BigNumber from "bignumber.js";
+import { expect } from "chai";
+import ActionType from "../enums/actionType";
+import Story from "../types/story";
+import ElyfiContracts from "../types/ElyfiContracts";
+import takeDataSnapshot from "./takeDataSnapshot";
+import { ContractTransaction } from "@ethersproject/contracts";
+import { ReserveData, UserData } from "../../test/utils/Interfaces";
+require('../../test/assertions/equals')
+
+const excuteInvestor = async (
+  account: Wallet,
+  amount: BigNumber,
+  elyfiContracts: ElyfiContracts,
+  doTransaction: () => Promise<ContractTransaction | undefined>,
+  calculateReserveData: (amount: BigNumber, reserveData: ReserveData, txTimestamp: BigNumber) => ReserveData,
+  calculateUserData: (amount: BigNumber, userData: UserData, reserveDateBefore: ReserveData, reserveDataAfter: ReserveData, txTimestamp: BigNumber) => UserData
+) => {
+  const [reserveDataBefore, userDataBefore] = await takeDataSnapshot(account, elyfiContracts)
+
+  const tx = await doTransaction();
```

Comment:
Could `waffle.provider.send('evm_mine', []);` change hre timestamp? 
We already implemented evm_mine method in utils.